src/components/fields/location-field/location-modal/location-picker/location-picker.tsx

```diff
@@ -174,7 +179,12 @@ export const LocationPicker = ({
 			const isSelected = enlargeSelectedMarker
 				? target.lat === selectedLocationCoord.lat && target.lng === selectedLocationCoord.lng
 				: undefined;
-			const marker = markerFrom(target, interactiveMapPinIconUrl, isSelected).addTo(map);
+			const mapPinIcon =
+				"data:image/svg+xml;base64," +
+				btoa(
+					ReactDOMServer.renderToString(<PinFillIcon color={Color.Primary({ theme: theme || BaseTheme })} />)
```

Comment:
similarly, you can apply theme through styled-components instead of retrieving it here

Reply:
the pin icon will be set and render before the styled-components apply the theme

Reply:
ok have tested, I am ok with the implementation.

long explanation: `ReactDOMServer.renderToString()` does not have themes because it is rendering completely separately from the render tree. passing as a prop is the easiest way since we are passing the theme from the "outside".